### DISCUSSION TITLE
Ability to define schema name for generated paginated response

### DISCUSSION THREAD

**Discussion by manuel-watchenterprise** [dedoc/scramble#606] [Ideas]
It can be personal taste to have everything in schemas but when it comes to code generators with hardly typed languages such as dart and java, it is almost required to have this option, otherwise ugly automatic names will appear in the application code. Also, it prevents duplications as multiple endpoints can return the same but all of them will have the full specification as it is not using schemas.

```php
/**
 * @response AnonymousResourceCollection<Paginator<UserResource>>
 * @schemaName UserListResponse
 */
public function __invoke(Request $request): AnonymousResourceCollection
{
   return UserResource::collection(
    $this->userService->getTeamMembers($request->input('per_page', 30))
  );
}
```

**Comment by manuel-watchenterprise** [comment]
Though I still believe this would be useful for the community, I've done this with our own extension. (not annotation but to automatically create / reuse typed paginated schema definitions.

**Reply by dti4kyba** [reply]
Can you share with us? Thanks
